stravalib/model.py

```diff
@@ -240,16 +357,49 @@ class LatLon(LatLng, BackwardCompatibilityMixin, DeprecatedSerializableMixin):
     """
 
     @root_validator
-    def check_valid_latlng(cls, values):
-        # Strava sometimes returns an empty list in case of activities without GPS
+    def check_valid_latlng(
+        cls, values: List[Optional[float]]
+    ) -> Optional[List[Optional[float]]]:
```

Comment:
I feel like here the types shoud be `list[float]` for input and `list[float] | None` for output unless I'm misunderstanding and the list themselves can contain None value?

Reply:
Ok so i *may* be reading this wrong but here is my understanding... there was a comment about the lat/lon values returned by Strava being empty sometimes (when say someone creates an activity but there is no gps attached). as such you could get an empty list here as an input to this method. 

i did not test this however (i'm still wrapping my head around our test suite) but i do know having activities with no x,y data is possible in strava.

Reply:
The input is either a list of exactly two elements or an empty list, so `list[float]` should be its type. AFAIK we cannot specify that lists have exactly _n_ elements (as we can for tuples, right?).

The same goes for the output, but it can be `None` if the input is an empty list. So its type would become `list[float] | None`

Reply:
ok so if i understand correctly @jsamoocha the input should be either 

`List[Optional[float]]` which would be the same as `list[float | None]` since it can be a empty list??

then the output would be 

Optional[list[float]]  which is `list[float] | None`

i'm going to make that change now but can change if i am not understanding correctly.

Reply:
I think `list[float]` is fine as input type since it can be empty. The type `list[float | None]` indicates that the list elements can be `None`, which is not the case here.

Reply:
ok thank you for the clarification. fixed! i think in my mind a None was the same as empty. but I understand now that they are two different things. (`None` vs empty list).  mypy is ok with this solution. resolving this